### PR TITLE
added splash screen, fav icon and app working without internet connection

### PR DESCRIPTION
**Pull request by LautaroSanchezGl**
Added Splash Screen
Added FavIcon
Adden app name natively
App working without internet connection

DEMO

https://github.com/LautaroSanchezGl/TheMovieDB/assets/142030231/c99133ab-0fed-4553-9aee-73ef51b5e243

### CODE REVIEW

File: lib/data/datasource/local/local_daos/genre_dao.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+@dao
+abstract class GenreDao{
+  
+  @Insert(onConflict: OnConflictStrategy.replace)
+  Future<void> addGenre(Genre genre);
+
+  @Query("Select * from Genre where id in (:ids)")
+  Future<List<Genre>> getGenreByIds(List<int> ids);
+  
+
```
Comment (MarcosCGlobant) [inline lib/data/datasource/local/local_daos/genre_dao.dart]: format code to remove extra line

File: lib/data/repository/movie_database_repository_imp.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  MovieDatabaseRepositoryImp(this._movieDatabase);
+
+  @override
+  Future<void> addMovie(movie) async {
+    _movieDatabase.movieDao.insertMovie(movie);
+  }
+
+  @override
+  Future<List<Movie>> getMovies({String category = ""}) async {
```
Comment (MarcosCGlobant) [inline lib/data/repository/movie_database_repository_imp.dart]: why do you pass the category as empty?

File: lib/data/datasource/local/local_daos/genre_dao.dart (hunk 1 above)
Comment (LautaroSanchezGl) [reply lib/data/datasource/local/local_daos/genre_dao.dart]: done

File: lib/data/repository/movie_database_repository_imp.dart (hunk 2 above)
Comment (LautaroSanchezGl) [reply lib/data/repository/movie_database_repository_imp.dart]: actually, cant remember.. but it was unnecessary. I made it required to fix it. Thanks!

File: lib/domain/usecase/implementation/get_movies_usecase.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            await movieDataBaseRepository.existsById(id: movie.id);
+        if (movieInDb != null) {
+          if (!movieInDb.categories.contains(endpoint.category)) {
+            movieInDb.categories.add(endpoint.category);
+            movieDataBaseRepository.addMovie(movieInDb);
+          }
+        } else {
+          movie.categories.add(endpoint.category);
+          movieDataBaseRepository.addMovie(movie);
+        }
```
Comment (FranciscoGaraventa) [inline lib/domain/usecase/implementation/get_movies_usecase.dart]: Maybe you can move the logic to validate if the movie exist or not on movie_database_repository

File: lib/main.dart [hunk 4]
```diff
@@ -2,22 +2,49 @@ import 'package:flutter/material.dart';
...
+
+import 'presentation/bloc/movie_genres_bloc.dart';
 import 'presentation/bloc/movies_bloc.dart';
 import 'presentation/view/movies_grid_view.dart';
+import 'package:provider/provider.dart';
+
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  final db =
+      await $FloorMovieDatabase.databaseBuilder('movie_database_v3').build();
```
Comment (FranciscoGaraventa) [inline lib/main.dart]: add db value to const

File: lib/domain/usecase/implementation/get_movie_genres_usecase.dart [hunk 5]
```diff
@@ -1,15 +1,44 @@
...
+
+  GetMovieGenresUseCase({
+    required this.genreDatabaseRepository,
+  });
 
   @override
-  Future<DataState<List<Genre>>> call({params}) {
-    return genreRepository.getMovieGenres(params);
+  Future<DataState<List<Genre>>> call({params}) async {
+    var connectivityResult = await (Connectivity().checkConnectivity());
```
Comment (FranciscoGaraventa) [inline lib/domain/usecase/implementation/get_movie_genres_usecase.dart]: You can remove unnecessary '()'

File: lib/domain/usecase/implementation/get_movie_genres_usecase.dart [hunk 6]
```diff
@@ -1,15 +1,44 @@
...
+    if (connectivityResult == ConnectivityResult.none) {
+      try {
+        List<Genre> genres =
+            await genreDatabaseRepository.getMovieGenres(params);
+        return DataSuccess(genres);
+      } on Exception catch (e) {
+        return DataFailed(e.toString());
```
Comment (FranciscoGaraventa) [inline lib/domain/usecase/implementation/get_movie_genres_usecase.dart]: Why do you need to have the try-catch block here

File: lib/domain/usecase/implementation/get_movies_usecase.dart (hunk 3 above)
Comment (LautaroSanchezGl) [reply lib/domain/usecase/implementation/get_movies_usecase.dart]: done

File: lib/main.dart (hunk 4 above)
Comment (LautaroSanchezGl) [reply lib/main.dart]: done

File: lib/domain/usecase/implementation/get_movie_genres_usecase.dart (hunk 5 above)
Comment (LautaroSanchezGl) [reply lib/domain/usecase/implementation/get_movie_genres_usecase.dart]: done

File: lib/domain/usecase/implementation/get_movie_genres_usecase.dart (hunk 6 above)
Comment (LautaroSanchezGl) [reply lib/domain/usecase/implementation/get_movie_genres_usecase.dart]: done

**Review by FranciscoGaraventa — APPROVED** [review]
LG

**Review by MarcosCGlobant — APPROVED** [review]

**Review by Kevin-Guillen — APPROVED** [review]

**Pull request merged**